### DISCUSSION TITLE
Can we declare renovate groups in a dependency's repository?

### DISCUSSION THREAD

**Discussion by deviantintegral** [renovatebot/renovate#25079] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Renovate ships with many default group presets at https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/group.ts.

For our projects, there's two common additions we add to each project:

Drupal:

```
    {
      "matchPackageNames": ["drupal/core", "drupal/core-recommended", "drupal/core-composer-scaffold", "drupal/core-project-message"],
      "groupName": "Drupal Core"
    },
```

Drainpipe:

```
    {
      "matchPackageNames": ["lullabot/drainpipe", "lullabot/drainpipe-dev"],
      "groupName": "Drainpipe packages"
    },
```

For Drupal, I could see adding it to Renovate itself since it's a large ecosystem. Yet for Drainpipe, there's dozens, not hundreds of thousands of installs. I imagine we could create a preset for it, but we'd still have to reference it from our configuration. Is there a way we can add a configuration file to the drainpipe repo that is _automatically_ used by renovate, with zero additional configuration from downstream users?

### What you tried so far.

Nothing - this is a question on concepts and best practices.

**Comment by rarkins** [comment]
Groupings which are part of `group:recommend` are included as part of `config:recommended`, so that would be the place to add them.

**Reply by deviantintegral** [reply]
Thanks. My understanding is that we could configure this centrally with a [Sharable Config preset](https://docs.renovatebot.com/config-presets/), and then import that into each project.

But is there a way for a public dependency to declare those groups for consumers? My understanding was that renovate.json was "root only" and never rolled up into the app hosting the dependency.

**Reply by rarkins** [reply, marked as answer]
There's no way for a public dependency to declare its groups
